### PR TITLE
Checkout progress

### PR DESCRIPTION
**Pull request by hugokallstrom**
### Checklist

- [ ] Functionality is covered by an integration test
- [ ] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/CheckoutActivity.kt [hunk 1]
```diff
@@ -36,6 +38,7 @@ class CheckoutActivity : BaseActivity(R.layout.activity_checkout) {
     }
     private val viewModel: CheckoutViewModel by viewModel { parametersOf(parameter.quoteIds) }
     private val binding by viewBinding(ActivityCheckoutBinding::bind)
+    private lateinit var progressDialog: AlertDialog
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/CheckoutActivity.kt:41]: How come we're making this `lateinit`? 🤔 Couldn't we create this where it's used?

File: app/src/main/res/layout/progress_dialog.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        android:layout_height="@dimen/loading_spinner_height"
+        android:layout_margin="@dimen/base_margin" />
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerVertical="true"
+        android:layout_margin="@dimen/base_margin"
+        android:layout_toEndOf="@+id/progressBar"
+        android:text="Signing..." />
```
Comment (hedvigoscar) [inline app/src/main/res/layout/progress_dialog.xml:21]: 👮

**Review by hedvigoscar — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/CheckoutActivity.kt (hunk 1 above)
Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/CheckoutActivity.kt:41]: Unfortunately not, we need the activity to be created before we can create the dialog

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/CheckoutActivity.kt:41]: I mean, can't we create it where we invoke it? That must be after `onCreate`, right? 🤔

Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/CheckoutActivity.kt:41]: We want to be able to dismiss the dialog if we receive an error, thats why I created it here and then just do show/dismiss when handling an event

File: app/src/main/res/layout/progress_dialog.xml (hunk 2 above)
Comment (hugokallstrom) [reply app/src/main/res/layout/progress_dialog.xml]: Good catch, created a string res 🙏

Comment (hedvigoscar) [reply app/src/main/res/layout/progress_dialog.xml]: Alright, fair enough.

File: app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/CheckoutActivity.kt (hunk 1 above)
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/CheckoutActivity.kt:41]: Alright, fair enough.

**Pull request merged**